docs/content/using-npm/config.md

```diff
@@ -753,6 +754,14 @@ Used with `npm ls`, limiting output to only those packages that are linked.
 The IP address of the local interface to use when making connections to the
 npm registry. Must be IPv4 in versions of Node prior to 0.12.
 
+#### `location`
+
+* Default: "user" unless `--global` is passed, which will also set this value
+  to "global"
+* Type: "global", "user", or "project"
```

Comment:
built-in?

Reply:
leaving that out was intentional, feels like the built-in use case should be a file overwritten by packagers and not something a user would interactively change on their own to me. I'm definitely open to discuss it though

Reply:
I’d expect it to be undocumented regardless, but it seems pretty useful for debugging the source of an unexpected default config (for “get”), and it seems weird to allow it for get and not for set.

Reply:
`get` and `list` both ignore the parameter entirely. they aren't location specific, they return whatever the most prioritized value(s) is/are. we could allow for the location flag on them as a means of saying "show me only data from this location" in which case I agree exposing built-in would be appropriate. if we did that, though, we would have to remove the default value for the location and have separate behaviors in the commands for a value of `null` which i don't love...

Reply:
ah hm, that's a fair point.

if it's only going to work in `set` then omitting built-in makes sense; but it seems more useful to me if it works for all of get/list/set, and includes built-in.

Reply:
to address the issue of debugging the source of an unexpected config value, how about we allow `npm config list` to accept config keys for which it will list everywhere that key is defined and what overwrote what.

so `npm config get key` will continue to return a single value representing the value of `key` from its highest prioritized source, and `npm config list -l key` will list every value that `key` has in any source in a way that makes it clear why it was overridden from other sources.

Reply:
For list, that sounds great.